File: 26.remove-duplicates-from-sorted-array.c
/*
 * @lc app=leetcode id=26 lang=c
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start


int removeDuplicates(int* nums, int numsSize)
{
    if (!nums || !numsSize) {
        return 0;
    }
    int last_index = 0;
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[last_index]) {
            nums[++last_index] = nums[i];
        }
    }
    return last_index + 1;
}
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.go
/*
 * @lc app=leetcode id=26 lang=golang
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
// 快慢指针
func removeDuplicates(nums []int) int {
	slow := 0
	for fast := 1; fast < len(nums); fast++ {
		// 如果不一致，把fast的值移到slow的下一个
		if nums[fast] != nums[slow] {
			slow++
			nums[slow] = nums[fast]
		}
	}
	return slow + 1
}

// @lc code=end
